pkg/controllers/multiclusterservice/mcs_controller.go

```diff
@@ -125,21 +130,57 @@ func (c *MCSController) handleMCSDelete(ctx context.Context, mcs *networkingv1al
 }
 
 func (c *MCSController) deleteServiceWork(mcs *networkingv1alpha1.MultiClusterService, retainClusters sets.Set[string]) error {
-	mcsID, err := c.getMultiClusterServiceID(mcs)
+	mcsID := util.GetLabelValue(mcs.Labels, networkingv1alpha1.MultiClusterServicePermanentIDLabel)
+	workList, err := helper.GetWorksByLabelsSet(c, labels.Set{
+		networkingv1alpha1.MultiClusterServicePermanentIDLabel: mcsID,
+		util.MultiClusterServiceOriginalTypeLabel:              util.ServiceTypeLabelValue,
+	})
 	if err != nil {
-		klog.Errorf("Get MultiClusterService(%s/%s) ID error:%v", mcs.Namespace, mcs.Name, err)
+		klog.ErrorS(err, "failed to get work", "namespace", mcs.Namespace, "name", mcs.Name)
 		return err
 	}
 
-	workList, err := helper.GetWorksByLabelsSet(c, labels.Set{networkingv1alpha1.MultiClusterServicePermanentIDLabel: mcsID})
+	for index := range workList.Items {
+		clusterName := strings.TrimPrefix(workList.Items[index].Namespace, names.ExecutionSpacePrefix)
+		if retainClusters.Has(clusterName) {
+			continue
+		}
+
+		if err = c.Client.Delete(context.TODO(), &workList.Items[index]); err != nil && !apierrors.IsNotFound(err) {
+			klog.Errorf("Error while updating work(%s/%s) deletion timestamp: %s",
```

Comment:
Log content mismatch with action. Here is trying to delete a work, not updating.